testing/e2e/e2e_withdrawal_test.go

```diff
@@ -0,0 +1,262 @@
+		credentials := validator.Validator.WithdrawalCredentials
+
+		// Create beacon-kit's WithdrawalCredentials type
+		var beaconKitCredentials ctypes.WithdrawalCredentials
+		copy(beaconKitCredentials[:], credentials)
+
+		// Check if it has execution withdrawal credentials
+		if beaconKitCredentials.IsValidEth1WithdrawalCredentials() {
+			validatorIndex = fmt.Sprintf("%d", validator.Index)
+			// Convert the phase0.BLSPubKey to our crypto.BLSPubkey type
+			copy(blsPubkey[:], validator.Validator.PublicKey[:])
+			return validatorIndex, blsPubkey, nil
+		}
+	}
+
+	return "", crypto.BLSPubkey{}, errors.New("no validator with execution withdrawal credentials found")
+}
+
+// TestSubmitPartialWithdrawalTransaction tests submitting a partial withdrawal transaction via the withdrawal contract
+func (s *BeaconKitE2ESuite) TestSubmitPartialWithdrawalTransaction() {
+	// Use timeout context to better control the test
+	ctx, cancel := context.WithTimeout(s.Ctx(), suite.DefaultE2ETestTimeout)
+	defer cancel()
+
+	// Initialize test client
+	client := s.initBeaconTest()
+
+	// Find a validator with execution credentials
+	validatorIndex, blsPubkey, err := s.findValidatorWithExecutionCredentials(client)
+	s.Require().NoError(err)
+	s.T().Logf("Found validator with index %s for withdrawal test", validatorIndex)
+
+	// Set withdrawal amount
+	withdrawalAmount := beaconmath.Gwei(DefaultWithdrawalAmount)
+
+	// Create an rpc client using the load balancer URL
+	rpcClient, err := rpc.Dial(s.JSONRPCBalancer().URL())
+	s.Require().NoError(err)
+	defer rpcClient.Close()
+
+	rpcWrapper := &rpcWrapper{Client: rpcClient}
+
+	// Get current block number before withdrawal
+	blkNum, err := s.JSONRPCBalancer().BlockNumber(ctx)
+	s.Require().NoError(err)
+	s.T().Logf("Block number before withdrawal: %d", blkNum)
+
+	// Check for pending partial withdrawals before submitting the transaction
+	pendingWithdrawalsBefore, err := s.getPendingPartialWithdrawals(utils.StateIDHead)
+	s.Require().NoError(err)
+	s.Require().Len(pendingWithdrawalsBefore, 0, "Expected no pending withdrawals initially")
+
+	// Get the withdrawal fee
+	fee, err := eip7002.GetWithdrawalFee(ctx, rpcWrapper)
+	s.Require().NoError(err)
+	s.T().Logf("Withdrawal fee: %s wei", fee.String())
+
+	// Create the withdrawal transaction data
+	withdrawalTxData, err := eip7002.CreateWithdrawalRequestData(blsPubkey, withdrawalAmount)
+	s.Require().NoError(err)
+
+	// Use a pre-loaded key that has funds
+	privateKey, err := ethcrypto.HexToECDSA("fffdbb37105441e14b0ee6330d855d8504ff39e705c3afa8f859ac9865f99306")
+	s.Require().NoError(err)
+
+	chainID, err := s.JSONRPCBalancer().ChainID(ctx)
+	s.Require().NoError(err)
+	signer := gethcore.NewPragueSigner(chainID)
+
+	// Get the sender's nonce
+	var nonce hexutil.Uint64
+	err = rpcClient.CallContext(ctx, &nonce, "eth_getTransactionCount",
+		common.HexToAddress("0x20f33ce90a13a4b5e7697e3544c3083b8f8a51d4"), "latest",
+	)
+	s.Require().NoError(err)
```

Comment:
The eth1 EOA wallet used to sign the tx (in this case `0x20f33ce90a13a4b5e7697e3544c3083b8f8a51d4`) **needs** to be the exact address derived from the validator's own withdrawal credentials. Since we start the kurtosis devnet for all validators with the same credentials, I would comment [that](https://github.com/berachain/beacon-kit/blob/main/kurtosis/src/nodes/consensus/beacond/launcher.star#L73) here why its okay to use this wallet.

Reply:
addressed. Added comments